### ISSUE TITLE
cant edit description on clusters created with rke templates

### ISSUE THREAD

**Issue by aemneina** [rancher/rancher#26754]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. create cluster and select an existing rke template
2. after cluster is provisioned, edit cluster, modifying description
3. click save

**Result:**
the description doesnt get updated or saved. custom clusters work fine, its just when you use rke templates, the description change is omitted.

**Comment by aemneina** [comment]
UI doesnt seem to preserve annotations added via the UI as well.

**Comment by soumyalj** [comment]
Reproduced the issue with v2.4.3. 
1. Created a RKE template "template1"
2. Create a cluster using "template1"
3. Edit the cluster and add description, labels and annotations and save
The description/labels/annotation do not get updated.

**Comment by soumyalj** [comment]
Validated the fix with master-head build (dea919fb2)

1. Create a RKE template "template1"
2. Create a cluster using "template1"
Edit the cluster and add description, labels and annotations and save
The description, labels and annotations are updated correctly.
API also reflects the description, labels, annotations. 
```
created": "2020-05-21T00:32:57Z",
"createdTS": 1590021177000,
"creatorId": "user-nd9k6",
"defaultClusterRoleForProjectMembers": null,
"defaultPodSecurityPolicyTemplateId": null,
"description": "testdesc",
"desiredAgentImage": null,
"desiredAuthImage": null,
"dockerRootDir": "/var/lib/docker",
"driver": "rancherKubernetesEngine",
"enableClusterAlerting": false,
"enableClusterMonitoring": false,
"enableNetworkPolicy": false,
"id": "c-zpq5b",
"internal": false,
"istioEnabled": false,
"labels": {
"cattle.io/creator": "norman",
"l1": "l2"
},

```
```
"annotations": {
"a1": "a2",
"authz.management.cattle.io/creator-role-bindings": "{\"created\":[\"cluster-owner\"],\"required\":[\"cluster-owner\"]}",
"lifecycle.cattle.io/create.cluster-agent-controller-cleanup": "true",
"lifecycle.cattle.io/create.cluster-provisioner-controller": "true",
"lifecycle.cattle.io/create.cluster-scoped-gc": "true",
"lifecycle.cattle.io/create.mgmt-cluster-rbac-remove": "true",
"provisioner.cattle.io/ke-driver-update": "updated"
},


```
